modules/hivemq/src/main/java/org/testcontainers/hivemq/HiveMQContainer.java

```diff
@@ -0,0 +1,651 @@
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Yannick Weber
+ */
+@SuppressWarnings("UnusedReturnValue")
+public class HiveMQContainer extends GenericContainer<HiveMQContainer> {
+
+    private final static @NotNull Logger logger = LoggerFactory.getLogger(HiveMQContainer.class);
+
+    private static final @NotNull String validPluginXML =
+            "<hivemq-extension>" + //
+                    "   <id>%s</id>" + //
+                    "   <name>%s</name>" + //
+                    "   <version>%s</version>" + //
+                    "   <priority>%s</priority>" +  //
+                    "   <start-priority>%s</start-priority>" +  //
+                    "</hivemq-extension>";
+
+    public static final @NotNull String DEFAULT_HIVEMQ_EE_TAG = "4.7.2";
+    public static final @NotNull String DEFAULT_HIVEMQ_CE_TAG = "2021.3";
+    public static final @NotNull DockerImageName DEFAULT_HIVEMQ_EE_IMAGE_NAME = DockerImageName.parse("hivemq/hivemq4").withTag(DEFAULT_HIVEMQ_EE_TAG);
+    public static final @NotNull DockerImageName DEFAULT_HIVEMQ_CE_IMAGE_NAME = DockerImageName.parse("hivemq/hivemq-ce").withTag(DEFAULT_HIVEMQ_CE_TAG);
+
+    public static final int DEBUGGING_PORT = 9000;
+    public static final int MQTT_PORT = 1883;
+    public static final int CONTROL_CENTER_PORT = 8080;
+    @SuppressWarnings("OctalInteger")
+    private static final int MODE = 0777;
+    private static final @NotNull Pattern EXTENSION_ID_PATTERN = Pattern.compile("<id>(.+?)</id>");
+    public static final @NotNull String EXTENSION_MAIN_CLASS_NAME = "com.hivemq.extension.sdk.api.ExtensionMain";
+
+    private final @NotNull ConcurrentHashMap<String, CountDownLatch> containerOutputLatches = new ConcurrentHashMap<>();
+    private volatile boolean silent = false;
+    private volatile boolean controlCenterEnabled = false;
+
+    private final @NotNull MultiLogMessageWaitStrategy waitStrategy = new MultiLogMessageWaitStrategy();
+
+    public HiveMQContainer() {
+        this(DEFAULT_HIVEMQ_CE_IMAGE_NAME.withTag(DEFAULT_HIVEMQ_CE_TAG));
+    }
+
+    public HiveMQContainer(final @NotNull DockerImageName dockerImageName) {
+        super(dockerImageName);
+
+        dockerImageName.assertCompatibleWith(DEFAULT_HIVEMQ_CE_IMAGE_NAME, DEFAULT_HIVEMQ_EE_IMAGE_NAME);
+
+        addExposedPort(MQTT_PORT);
+
+        waitStrategy.withRegEx("(.*)Started HiveMQ in(.*)");
+        waitingFor(waitStrategy);
+
+        withLogConsumer(outputFrame -> {
+            final String utf8String = outputFrame.getUtf8String();
+            if (utf8String.startsWith("Listening for transport dt_socket at address:")) {
+                System.out.println("Listening for transport dt_socket at address: " + getMappedPort(DEBUGGING_PORT));
```

Comment:
The reason why we chose stdout is: HiveMQ itself also uses logback, we want to avoid getting the logback prefix two times, in order to preserve readability. 

Is there any other way we can achieve this?

Reply:
In this particular line we need sout, because we need to intercept the "Listening for transport dt_socket at address: " log from the JVM and change the port with the mapped port, in case `withDebugging()` is used. 

This exact log message is acknowledged by some IDEs that present you with the option of attaching a debugger after seeing this log. 

I got rid of the other souts :)

Reply:
@rnorth Can we see this as resolved?

Reply:
Yes makes sense, I think the IDE integration like this is pretty neat 🙂

Reply:
A number of our users have expressed concerts about the logs that cannot be suppressed. Perhaps we should make it optional? I do not see many folks needing to debug HiveMQ's process, at least not as a day-to-day thing 😅